zkvm/src/vm.rs

```diff
@@ -292,16 +292,16 @@ where
     }
 
     fn neg(&mut self) -> Result<(), VMError> {
-        let expr = self.pop_item()?.to_expression()?;
-        let neg_expr = Expression {
-            terms: expr
-                .terms
-                .iter()
-                .map(|t| (t.0, t.1.neg()))
-                .collect::<Vec<_>>(),
-            assignment: expr.assignment,
-        };
-        self.push_item(neg_expr);
+        let mut expr = self.pop_item()?.to_expression()?;
```

Comment:
`Neg::neg` consumes the value, is `mut` necessary here?
 
```suggestion
        let expr = self.pop_item()?.to_expression()?;
```